### PR TITLE
APIO-244 Page operations are back!

### PR DESCRIPTION
**Pull request by ahdezma**
Important points of this PR: 

- Added missing tests and Javadocs to `ActionManager` utils.
- Fixes some errors that were occurring in the `ANNOTATIONS` branch: https://github.com/liferay/com-liferay-apio-architect/commit/89949f41f4e849890ede7dc446e1e7cdac78f827 and https://github.com/liferay/com-liferay-apio-architect/commit/1e0e16bed50dfd5d10ba1d842edb7d645c0a518f
- Make `Resource.Id` API, so it can be returned on `Page` and `SingleModel` instances, simplifying URL generation.
- Bring back page operations 🎉 🎉

### CODE REVIEW

File: apio-architect-impl/src/test/java/com/liferay/apio/architect/internal/action/ActionSemanticsTest.java [hunk 1]
```diff
@@ -283,9 +271,9 @@ public void testWithNameReturnsActionSemanticsWithDifferentName() {
 		ActionSemantics newActionSemantics = actionSemantics.withName("create");
 
 		assertEquals(actionSemantics, actionSemantics.withName("retrieve"));
-		assertThat(actionSemantics.name(), is("retrieve"));
+		assertThat(actionSemantics.getActionName(), is("retrieve"));
```
Comment (nhpatt) [inline apio-architect-impl/src/test/java/com/liferay/apio/architect/internal/action/ActionSemanticsTest.java]: Why did you change all methods to get_

File: apio-architect-impl/src/test/java/com/liferay/apio/architect/internal/annotation/util/ActionRouterUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+			"withListBodyParameters", List.class);
+		Method singleBodyMethod = MyAnnotatedInterface.class.getMethod(
+			"withIdAndBodyParameters", Pagination.class, Credentials.class,
+			Long.class, MyType.class);
+		Method notAnnotatedMethod = MyAnnotatedInterface.class.getMethod(
+			"notAnnotated");
+
+		assertTrue(isListBody(listBodyMethod));
+		assertFalse(isListBody(singleBodyMethod));
+		assertFalse(isListBody(notAnnotatedMethod));
```
Comment (nhpatt) [inline apio-architect-impl/src/test/java/com/liferay/apio/architect/internal/annotation/util/ActionRouterUtilTest.java]: sort :P

File: apio-architect-impl/src/test/java/com/liferay/apio/architect/internal/annotation/util/AnnotationUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+/**
+ * @author Alejandro Hernández
+ * @author Javier Gamarra
+ */
+public class AnnotationUtilTest {
+
+	@BeforeClass
+	public static void setUpClass() throws NoSuchMethodException {
+		_annotatedWithRemoveMethod = MyAnnotatedInterface.class.getMethod(
```
Comment (nhpatt) [inline apio-architect-impl/src/test/java/com/liferay/apio/architect/internal/annotation/util/AnnotationUtilTest.java]: sort?

File: apio-architect-impl/src/test/java/com/liferay/apio/architect/internal/annotation/util/ActionRouterUtilTest.java (hunk 2 above)
Comment (ahdezma) [reply apio-architect-impl/src/test/java/com/liferay/apio/architect/internal/annotation/util/ActionRouterUtilTest.java]: https://github.com/liferay/com-liferay-apio-architect/pull/320/commits/ecd917cbd811e9cd7f5a274b383fd68265341da7

File: apio-architect-impl/src/test/java/com/liferay/apio/architect/internal/annotation/util/AnnotationUtilTest.java (hunk 3 above)
Comment (ahdezma) [reply apio-architect-impl/src/test/java/com/liferay/apio/architect/internal/annotation/util/AnnotationUtilTest.java]: https://github.com/liferay/com-liferay-apio-architect/pull/320/commits/ecd917cbd811e9cd7f5a274b383fd68265341da7

File: apio-architect-impl/src/main/java/com/liferay/apio/architect/internal/annotation/ActionManagerImpl.java [hunk 4]
```diff
@@ -315,7 +315,7 @@ public EntryPoint getEntryPoint() {
 		);
 	}
 
-	private Object _getBody(HttpServletRequest request) {
+	private Body _getBody(HttpServletRequest request) {
```
[the remark this answers is not in the document]
Comment (ahdezma) [reply apio-architect-impl/src/main/java/com/liferay/apio/architect/internal/annotation/ActionManagerImpl.java]: `getBodyValue` returns the final object (`BlogPosting`, `Comment`) created with the `Proxy`. But `getBody` always returns a `Body` (both `jsonToBody` and `multipartToBody` returns a `Body`)

File: apio-architect-impl/src/main/java/com/liferay/apio/architect/internal/annotation/util/AnnotationUtil.java [hunk 5]
```diff
@@ -14,65 +14,45 @@
...
-				Class::getAnnotations
-			).flatMap(
-				Stream::of
-			).filter(
-				annotationClass::isInstance
-			).map(
-				annotationClass::cast
-			).findFirst()
-		);
+		return null;
```
[the remark this answers is not in the document]
Comment (ahdezma) [reply apio-architect-impl/src/main/java/com/liferay/apio/architect/internal/annotation/util/AnnotationUtil.java]: > Each thing is for its thing

_@p2kmgcl's mum_
